doc/man7/ossl-guide-libssl-introduction.pod

```diff
@@ -0,0 +1,114 @@
+B<SSL_CTX>.
+
+An B<SSL> object is used to represent a connection to a remote peer. Once a
+connection with a remote peer has been established data can be exchanged with
+that peer.
+
+When using DTLS any data that is exchanged uses "datagram" semantics, i.e.
+the packets of data can be delivered in any order, and they are not guaranteed
+to arrive at all. In this case the B<SSL> object used for the connection is also
+used for exchanging data with the peer.
+
+Both TLS and QUIC support the concept of a "stream" of data. Data sent via a
+stream is guaranteed to be delivered in order without any data loss. A stream
+can be uni- or bi-directional.
+
+SSL/TLS only supports one stream of data per connection and it is always
+bi-directional. In this case the B<SSL> object used for the connection also
+represents that stream. See L<ossl-guide-tls-introduction(7)> for more
+information.
+
+The QUIC protocol can support multiple streams per connection and they can be
+uni- or bi-directional. In this case an B<SSL> object can represent the
+underlying connection, or a stream, or both. Where multiple streams are in use
+a separate B<SSL> object is used for each one. See
+L<ossl-guide-quic-introduction(7)> for more information.
+
+An B<SSL_CTX> object is used to create the B<SSL> object for the underlying
+connection. A single B<SSL_CTX> object can be used to create many connections
+(each represented by a separate B<SSL> object). Many API functions in libssl
+exist in two forms: one that takes an B<SSL_CTX> and one that takes an B<SSL>.
+Typically settings that you apply to the B<SSL_CTX> will then be inherited by
+any B<SSL> object that you create from it. Alternatively you can apply settings
+directly to the B<SSL> object without affecting other B<SSL> objects. Note that
+you should not normally make changes to an B<SSL_CTX> after the first B<SSL>
+object has been created from it.
+
+=head1 DATA STRUCTURES
+
+As well as B<SSL_CTX> and B<SSL> there are a number of other data structures
+that an application may need to use. They are summarised below.
+
+=over 4
+
+=item B<SSL_METHOD> (SSL Method)
+
+This structure is used to indicate the kind of connection you want to make, e.g.
+whether it is to represent the client or the server, and whether it is to use
+SSL/TLS, DTLS or QUIC. It is passed as a parameter when creating the B<SSL_CTX>.
+
+=item B<SSL_SESSION> (SSL Session)
+
+After establishing a connection with a peer the agreed cryptographic material
+can be reused to create future connections with the same peer more rapidly. The
+set of data used for such a future connection establishment attempt is collected
+together into an B<SSL_SESSION> object. A single successful connection with a
+peer may generate zero or more such B<SSL_SESSION> objects for use in future
+connection attempts.
+
+=item B<SSL_CIPHER> (SSL Cipher)
+
+During connection establishment the client and server agree which cryptographic
```

Comment:
agree which ?

Reply:
I didn't understand what you wanted me to change here - so no change made.

Reply:
The sentence does not sound right..

Reply:
agree apon maybe?